### PR TITLE
update test packages and global.json to net8.0 rc.2

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes
Updates test packages to rc.2 and global.json min version.

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
     "rollForward": "latestFeature",
-    "version": "8.0.100-rc.1.23463.5"
+    "version": "8.0.0-rc.2.23480.2"
```
Comment (utpilla) [inline global.json:4]: @vishweshbankwar Could you update this to `8.0.100-rc.2.23502.2`?

Comment (utpilla) [reply global.json:4]: It looks like this `8.0.0-rc.2.23480.2` is latest AspNetCore runtime version and not the latest SDK version:

image

Comment (vishweshbankwar) [reply global.json:4]: My bad. I copied it from runtime folder instead of SDK. fixed in https://github.com/open-telemetry/opentelemetry-dotnet/pull/5002

Comment (reyang) [reply global.json:4]: > It looks like this `8.0.0-rc.2.23480.2` is latest AspNetCore runtime version and not the latest SDK version:
> 
> image

👍

**Pull request merged**